extensions/NV/SPV_NV_shader_invocation_reorder.asciidoc

```diff
@@ -0,0 +1,1241 @@
+ +
+ Returns the address of shader record buffer for the hit or miss record encoded in hit object. +
+ +
+ 'Result' is the address of data in shader record as encoded in the hit object. +
+ +
+ 'Result Type' must be a 32-bit 'integer type' 2-component vector. +
+ +
+ 'Hit Object' is a pointer to the hit object. +
+ +
+ This instruction is allowed only in *RayGenerationKHR*, *ClosestHitKHR* and *MissKHR* execution models. +
+1+|Capability: +
+*ShaderInvocationReorderNV*
+| 4 | 5257
+| '<id>' 'Result Type'
+| '<id>' 'Result' 
+| '<id>' 'Hit Object'
+|======
+
+
+[cols="5*1"]
+|======
+4+|[[OpHitObjectIsEmptyNV]]*OpHitObjectIsEmptyNV* +
+ +
+ Returns a boolean indicating whether the hit object is an empty hit object. +
+ +
+ 'Result' is true if hit object encodes a NOP, false otherwise. +
+ +
+ 'Result Type' must be a 'boolean type' scalar. +
+ +
+ 'Hit Object' must be a pointer to hit object. +
+ +
+ This instruction is allowed only in *RayGenerationKHR*, *ClosestHitKHR* and *MissKHR* execution models. +
+1+|Capability: +
+*ShaderInvocationReorderNV*
+| 3 | 5276
+| '<id>' 'Result Type'
+| '<id>' 'Result'
+| '<id>' 'Hit Object'
+|======
+
+
+[cols="5*1"]
+|======
+4+|[[OpHitObjectIsHitNV]]*OpHitObjectIsHitNV* +
+ +
+ Returns a boolean indicating whether the hit object has encoded a hit. +
+ +
+ 'Result' is true if hit object encodes a hit, false otherwise. +
+ +
+ 'Result Type' must be a 'boolean type' scalar. +
+ +
+ 'Hit Object' must be a pointer to hit object. +
+ +
+ This instruction is allowed only in *RayGenerationKHR*, *ClosestHitKHR* and *MissKHR* execution models. +
+1+|Capability: +
+*ShaderInvocationReorderNV*
+| 3 | 5277
+| '<id>' 'Result Type'
+| '<id>' 'Result'
+| '<id>' 'Hit Object'
+|======
+
+
+[cols="5*1"]
+|======
+4+|[[OpHitObjectIsMissNV]]*OpHitObjectIsMissNV* +
+ +
+ Returns a boolean indicating whether the hit object has encoded a miss. +
+ +
+ 'Result' is true if hit object encodes a miss, false otherwise. +
+ +
+ 'Result Type' must be a 'boolean type' scalar. +
+ +
+ 'Hit Object' must be a pointer to hit object. +
+ +
+ This instruction is allowed only in *RayGenerationKHR*, *ClosestHitKHR* and *MissKHR* execution models. +
+1+|Capability: +
+*ShaderInvocationReorderNV*
+| 3 | 5278
+| '<id>' 'Result Type'
+| '<id>' 'Result'
+| '<id>' 'Hit Object'
+|======
+--
+
+--
+Validation Rules
+----------------
+
+An OpExtension must be added to the SPIR-V for validation layers to check
+legal use of this extension:
+
+----
+OpExtension "SPV_NV_shader_invocation_reorder"
+----
+
+Interactions with SPV_KHR_ray_tracing
+-------------------------------------
+
+This extension requires SPV_KHR_ray_tracing
+
+
+Issues
+------
+
+None
+
+Revision History
+----------------
+
+[cols="5,15,15,70"]
+[grid="rows"]
+[options="header"]
+|========================================
+|Rev|Date|Author|Changes
+|1 |2022-08-01 |Ashwin Lele         | Internal revisions
+|========================================
+
```

Comment:
Please build the asciidoc and include the .html in the PR

Link the html from the top-level README.md

Reply:
added in lasted pushes.